src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,394 @@
+                            .setTimestamp(epochMicros)
+                            .setMessage(rfc5424Frame.msg.toString())
+                            .setDirectory(rfc5424Frame.structuredData.getValue(teragrepDirectory).toString())
+                            .setStream(rfc5424Frame.structuredData.getValue(teragrepStreamName).toString()) // Or is sourcetype/stream supposed to be rfc5424Frame.appName.toString() instead?
+                            .setHost(rfc5424Frame.hostname.toString())
+                            .setInput(new String(source, StandardCharsets.UTF_8))
+                            .setPartition(recordOffsetObject.getPartition().toString())
+                            .setOffset(recordOffsetObject.getOffset())
+                            .setOrigin(new String(origin, StandardCharsets.UTF_8))
+                            .build();
+
+
+                    // Calculate the size of syslogRecord that is going to be written to syslogAvroWriter-file.
+                    long capacity = syslogRecord.toByteBuffer().capacity();
+                    // handle initial timestamp
+                    if (epochMicros_last == 0L) {
+                        epochMicros_last = epochMicros;
+                    }
+                    // Check if there is still room in syslogAvroWriter for another syslogRecord. Commit syslogAvroWriter to HDFS if no room left, emptying it out in the process.
+                    // boolean fileCommitted = committedToHdfs(approximatedSize + capacity, lastObject); // FIXME: approximatedSize is not working properly without the use of flush() after append. File sizes are all over the place.
+                    boolean fileCommitted = committedToHdfs(syslogAvroWriter.getFileSize() + capacity, (RecordOffset) lastObject);
+                    if (fileCommitted) {
+                        LOGGER.debug("Target file size reached, file <{}> stored to <{}> in HDFS", syslogFile.getName(), lastObject.getTopic()+"/"+lastObject.getPartition()+"."+lastObject.getOffset());
+                    }else {
+                        LOGGER.debug("Target file size not yet reached, continuing writing records to <{}>.", syslogFile.getName());
+                    }
+                    // if more records can be inserted, update epochMicros_last with the timestamp of the last inserted record.
+                    epochMicros_last = epochMicros;
+                    // Add syslogRecord to syslogAvroWriter which has room for new syslogRecord.
+                    syslogAvroWriter.write(syslogRecord);
+//                    approximatedSize += capacity;
+                    lastObject = recordOffsetObject;
+                    // The difference between actual and approximate file size is about 2,4 % with 64M files. So setting the MaximumFileSize to 95 % of the target should make things work.
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
please do use UncheckedIOException if necessary

Reply:
Solved in commit 75cb6183b3d3a3e65c47733727ac2a6d92c3b089